src/screens/CategoriesFilterScreen/ListItem.js

```diff
@@ -0,0 +1,140 @@
+// @flow
+import React, { Component } from "react";
+import {
+  StyleSheet,
+  TouchableOpacity,
+  View,
+  Animated,
+  Dimensions,
+  Image
+} from "react-native";
+import Text from "../../components/Text";
+import { whiteColor, blackColor } from "../../constants/colors";
+import whiteCheck from "../../../assets/images/whiteCheck.png";
+import blackCheck from "../../../assets/images/blackCheck.png";
+import type { EventCategory } from "../../data/event";
+
+type ListItemProps = {
+  category: EventCategory,
+  onPress: Function,
+  selected: boolean
+};
+
+class ListItem extends Component<ListItemProps, { textWidth: number }> {
+  static defaultProps = {
+    onPress: () => {}
+  };
+
+  constructor(props: ListItemProps) {
+    super(props);
+
+    // No initial animation
+    Animated.timing(this.decorationWidth, {
+      toValue: Number(props.selected),
+      duration: 0,
+      useNativeDriver: true
+    }).start();
+
+    this.state = {
+      textWidth: 16 - Dimensions.get("window").width
+    };
+  }
+
+  componentWillReceiveProps(props: ListItemProps) {
+    Animated.timing(this.decorationWidth, {
+      toValue: Number(props.selected),
+      duration: 100,
+      useNativeDriver: true
+    }).start();
+  }
+
+  decorationWidth = new Animated.Value(0);
+
+  // Measuring text width on initial render
+  handleOnLayout = (event: { nativeEvent: { layout: { width: number } } }) =>
+    this.setState({ textWidth: event.nativeEvent.layout.width });
+
+  render() {
+    const { category, onPress, selected } = this.props;
+    const { textWidth } = this.state;
+    const textColor = selected && category.contrast ? blackColor : whiteColor;
+
+    return (
+      <TouchableOpacity
```

Comment:
Can we use our `Touchable` component here for accessibility defaults and native styling on Android?